tools/HogEdit/HogEdit.cpp

```diff
@@ -0,0 +1,339 @@
+      extractor(it);
+  } else {
+    for (const auto &i : file_path_list) {
+      if (auto it = find_file_entry(hog_header, i.string()); it != std::end(hog_header))
+        extractor(it);
+      else
+        std::cout << "File " << i << " is not in the hog file. Skipping..." << std::endl;
+    }
+  }
+}
+} // namespace D3
+
+static inline void get_hog_path(int &parsed_parguments, const int argc, char *const argv[]) {
+  if (parsed_parguments++ < argc)
+    D3::archive_path = argv[parsed_parguments];
+  else
+    throw std::invalid_argument("No hog file specified.");
+}
+
+static inline void get_base_path(int &parsed_parguments, const int argc, char *const argv[]) {
+  if (parsed_parguments++ < argc)
+    D3::base_path = argv[parsed_parguments];
+  else
+    throw std::invalid_argument("Missing directory argument");
+}
+
+static inline void get_file_path_list(int &parsed_parguments, const int argc, char *const argv[]) {
+  if (parsed_parguments++ < argc) {
+    std::filesystem::path txt_path = argv[parsed_parguments];
+    if (D3::base_path.empty())
+      D3::base_path = txt_path.parent_path();
+
+    std::ifstream input(txt_path, std::ios_base::in);
+    for (std::string line; std::getline(input, line);)
+      D3::file_path_list.push_back(line);
+  } else
+    throw std::invalid_argument("Missing filename argument");
+}
+
+void print_usage(int exit_value) {
+  std::cout << "HogEdit v" << HOGEDIT_MAJORVER << "." << HOGEDIT_MINORVER << "." << HOGEDIT_BUILD << "-g"
+            << HOGEDIT_GIT_HASH << std::endl
+            << R"(Usage:
+  hogedit <command> hog_archive [options] [files]
+
+  commands:
+    --create      Create a new hog file that will contain the specified files.
+                  * The directory option specifies the base directory of the specified files.
+                  * Will overwrite an existing hog archive.
+
+    --add         Add one or more files to an existing hog file.
+                  * The directory option specifies the base directory of the specified files.
+
+    --delete      Delete one or more files from a hog file.
+                  * Filenames are those listed in the hog file.
+
+    --list        Print a list of filenames of files contained in the hog file.
+
+    --extract     Extract the specified list of files from the hog file.
+                  * If no filenames are specified then all files will be extracted.
+                  * The directory option specifies extraction location.
+
+  options:
+    --directory   Specifies the directory for the command to use.
+                  * If not specified then it is implied to be the current directory.
+
+    --files-from  Load the list of files from a text file instead the command line.
+                  * Format must be one filepath per line.
+                  * If the directory option is not specified then the filepaths will be relative
+                    to the path of this file.
+)";
+  std::exit(exit_value);
```

Comment:
A `--set-timestamp` override would be helpful, then the creation is deterministic and could use the a git timestamp from the repo.

Reply:
The timestamp is taken from the file the last time the file was written to. It's already deterministic.